### PR TITLE
Replace codefmtlib with a very small Vimscript.

### PR DESCRIPTION
**Pull request by malcolmr**
After google/vim-codefmt#34, codefmt uses Maktaba directly for registration, and so codefmtlib is not needed.  This change turns codefmtlib into a small stub library that merely forwards calls to Maktaba (where possible).

### CODE REVIEW

File: autoload/codefmtlib.vim [hunk 1]
```diff
@@ -52,63 +47,30 @@ endif
...
-
-  " TODO(dbarnett): Check types.
-
-endfunction
-
-""
 " Registers {formatter} as a supported formatter. See @dict(Formatter) for the
 " API {formatter} must implement.
+"
+" This method is deprecated. Callers should replace calls to this with the
```
Comment (dbarnett) [inline autoload/codefmtlib.vim]: We have `@deprecated` for this purpose. Don't know how it works for multiple lines, but maybe you can condense it?

Comment (malcolmr) [inline autoload/codefmtlib.vim]: Sadly, setting `@deprecated` apparently just has the effect of deleting the function definition from the generated documentation. I'm not sure we actually want that.

Comment (dbarnett) [inline autoload/codefmtlib.vim]: Oh, cool! Maybe we should rethink that…
Anyway, thanks for checking.

**Comment by dbarnett** [comment]
LGTM.

I may not get to reviewing the corresponding google/vim-codefmt#34 until EoD MTV.

**Comment by malcolmr** [comment]
No problem. I'm going to hold off committing this until after that one, but there's no hurry.

**Pull request merged**